dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -405,4 +414,44 @@ public static long countFailures(List<ArtifactResults> table, String columnName)
         .filter(row -> row.getResult(columnName) == null || row.getFailureCount(columnName) > 0)
         .count();
   }
+
+  /**
+   * Returns mapping from jar file to simplified description of its {@link DependencyPath}s. A
+   * simplified description of {@link DependencyPath}s explains a common pattern ({@code
+   * groupId:artifactId}) in the path elements, helping to avoid listing more than 5 repetitive
+   * items in BOM dashboard.
+   */
+  private static ImmutableMap<String, String> simplifyDependencyPathMessage(
+      Multimap<Path, DependencyPath> jarToDependencyPaths) {
```

Comment:
since we care about order (I think) we should probably declare this as ImmutableMultimap

Reply:
FYI, per javadoc: "Instead of using the Multimap interface directly, prefer the subinterfaces ListMultimap and SetMultimap. These take their names from the fact that the collections they return from get behave like (and, of course, implement) List and Set, respectively."

https://google.github.io/guava/releases/23.3-jre/api/docs/com/google/common/collect/Multimap.html